exp/controllers/azuremachinepool_controller.go

```diff
@@ -269,6 +269,9 @@ func (ampr *AzureMachinePoolReconciler) reconcileNormal(ctx context.Context, mac
 	defer done()
 
 	log.Info("Reconciling AzureMachinePool")
+
+	machinePoolScope.SetInfrastructureMachineKind()
```

Comment:
Nit: it might be slightly better to call this in the machinepool scope `Close()` function since we don't actually patch the AzureMachinePool object until then, and that will ensure it always gets called before the function returns (regardless of what gets added in the future in `reconcileNormal`)

Reply:
@CecileRobertMichon I'm running into an issue where when I move it into `Close()`, it fails the unit tests as it can't patch when the Cluster is paused. Is there a cleaner way to do this, since we don't have access to the Cluster in `Close()` to see if it's paused or not.

Reply:
Are you saying it _should_ patch it when paused, or it _shouldn't_ patch it when paused? if it's the latter, should we just return before the defer if the machinepool is paused?

Can you point me to the unit test you're referring to ?

Reply:
To clarify, it should *not* patch when the Cluster is paused. Currently, [here](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/e2bd59253c859b151ac6f3222ad6c749e2a1ba75/exp/controllers/azuremachinepool_controller.go#L255) we have the defer `reconcilePause()` since the intention, I believe, is to always be closing the scope we create.

Reply:
[This](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/434c140926cbc4f88a66a6f1365ce0b4b54e9fc1/exp/controllers/azuremachinepool_controller_test.go#L62) is the test that's failing with an error saying we can't patch as the AzureMachinePool is not found. We can probably fix that error, but from a design POV I think it makes sense that we wouldn't want to silently update the status if the Cluster is paused.